cove/views.py

```diff
@@ -2,609 +2,30 @@
-
-
-def get_fields_present(*args, **kwargs):
-    counter = collections.Counter()
-    counter.update(fields_present_generator(*args, **kwargs))
-    return dict(counter)
-
-
-def schema_dict_fields_generator(schema_dict):
-    if 'properties' in schema_dict:
-        for property_name, value in schema_dict['properties'].items():
-            if 'oneOf' in value:
-                property_schema_dicts = value['oneOf']
-            else:
-                property_schema_dicts = [value]
-            for property_schema_dict in property_schema_dicts:
-                property_type_set = get_property_type_set(property_schema_dict)
-                if 'object' in property_type_set:
-                    for field in schema_dict_fields_generator(property_schema_dict):
-                        yield '/' + property_name + field
-                elif 'array' in property_type_set:
-                    fields = schema_dict_fields_generator(property_schema_dict['items'])
-                    for field in fields:
-                        yield '/' + property_name + field
-                yield '/' + property_name
-
-
-def get_schema_fields(schema_filename):
-    r = requests.get(schema_filename)
-    return set(schema_dict_fields_generator(jsonref.loads(r.text, object_pairs_hook=OrderedDict)))
-
-
-def get_counts_additional_fields(schema_url, json_data, context, current_app):
-    fields_present = get_fields_present(json_data)
-    schema_fields = get_schema_fields(schema_url)
-    data_only_all = set(fields_present) - schema_fields
-    data_only = set()
-    for field in data_only_all:
-        parent_field = "/".join(field.split('/')[:-1])
-        # only take fields with parent in schema (and top level fields)
-        # to make results less verbose
-        if not parent_field or parent_field in schema_fields:
-            print(field, current_app)
-            if '_' in field.split('/')[-1] and current_app == 'cove-ocds':
-                context['language_additional_field_warning'] = True
-            data_only.add(field)
-
-    return [('/'.join(key.split('/')[:-1]), key.split('/')[-1], fields_present[key]) for key in data_only]
-
-
-def datetime_or_date(instance):
-    result = strict_rfc3339.validate_rfc3339(instance)
-    if result:
-        return result
-    return datetime.datetime.strptime(instance, "%Y-%m-%d")
-
-
-def get_schema_validation_errors(json_data, schema_url, current_app):
-    schema = requests.get(schema_url).json()
-    validation_errors = collections.defaultdict(list)
-    format_checker = FormatChecker()
-    if current_app == 'cove-360':
-        format_checker.checkers['date-time'] = (datetime_or_date, ValueError)
-    for n, e in enumerate(validator(schema, format_checker=format_checker).iter_errors(json_data)):
-        validation_errors[e.message].append("/".join(str(item) for item in e.path))
-    return dict(validation_errors)
-    
-
-class CoveInputDataError(Exception):
+class CoveWebInputDataError(CoveInputDataError):
```

Comment:
I don't like this solution, but I'm not sure how else to achieve the same end result with a more generic exception